docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,850 @@
+with thread names prefixed as `inject-event-manager-`.
+
+=== Asynchronous Event Producer
+All asynchronous event producers must use the `Event.Emitter.emitAsync(..)` method instead of the synchronous `Event.Emitter.emit(..)`.
+
+The `emitAsync` method returns a `CompletionStage<MyEvent>` instance. When executed, it completes once all event observers have been submitted to the executor service. However, there is no guarantee that any event has been delivered—it may have been sent to anywhere from 0 to n observers (where n represents the number of synchronous observers).
+
+[source,java]
+.Asynchronous Event Producer Example
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_4, indent=0]
+----
+
+=== Asynchronous Observer
+Asynchronous observer methods are invoked from separate threads (through the executor service mentioned above), and their results are ignored by the Event Emitter; if there is an exception thrown from the observer method, it is logged with `WARNING` log level into logger named link:{service-registry-base-url}/io/helidon/service/registry/EventManager.html[`EventManager`].
+
+To declare an asynchronous observer use annotation
+link:{service-registry-base-url}/io/helidon/service/registry/Event.AsyncObserver.html[`@Event.AsyncObserver`]
+instead of
+link:{service-registry-base-url}/io/helidon/service/registry/Event.Observer.html[`@Event.Observer`].
+
+[source,java]
+.Asynchronous Event Observer Example
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_5, indent=0]
+----
+
+== Programmatic Lookup
+
+There are two primary ways to access services from the service registry:
+
+- *Dependency Injection* – Services are injected automatically at defined injection points.
+- *Programmatic Lookup* – Services are retrieved manually by accessing the service registry directly.
+
+If you want to use programmatic lookup, there are several ways how to get a
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`] instance:
+
+* Get the global
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`] via
+link:{service-registry-base-url}/io/helidon/service/registry/GlobalServiceRegistry.html[`GlobalServiceRegistry`]
+** This shouldn’t be used from the factories or services.
+Intended use is in the `main` methods or when one needs static access.
+* Create a new
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`]
+instance and search for the desired service.
+* Inject the
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`]
+instance which is used to create the service instance.
+
+To create a registry instance:
+
+[source,java]
+----
+include::{sourcedir}/se/inject/ServiceRegistryExample.java[tag=snippet_1, indent=0]
+----
+
+Keep in mind, that
```

Comment:
I would also add a note here that in most cases, the single, global service registry is sufficient (and expected to be used) for a single application.
A custom service registry should only be created for specific use cases.
When I read this, it seems the three options are equal for use, but that is not the case. 
The default use case is to use `Services` to access the registry, or use the `GlobalServiceRegistry` for more fine grained use cases.
Creating a new registry manager is only intended for custom Main classes, or advanced use cases. It should be clear here, as otherwise people will end up creating multiple registries and it will not work as expected.